File: Graphs/graph implementation.cpp


//-------------------------------------------------------------------------------------------
//----------------------------------- unweighted graph --------------------------------------
//-------------------------------------------------------------------------------------------

int n, m;
vector<vector<int> > g;

int main() {
	scanf("%d%d", &n, &m);

	g.clear();
	g.resize(n);

	for (int i = 0; i < m; i++) {
		int a, b;
		scanf("%d%d", &a, &b);

		g[a].push_back(b); // a -> b
		g[b].push_back(a); // b -> a (for un-directed graph)
	}

	return 0; 
}



//-------------------------------------------------------------------------------------------
//------------------------------------- weighted graph --------------------------------------
//-------------------------------------------------------------------------------------------

int n, m;
vector<vector<pair<int,int> > > g;

int main() {

	scanf("%d%d", &n, &m);

	g.clear();
	g.resize(n);

	for (int i = 0; i < m; i++) {
		int a, b, c;
		scanf("%d%d%d", &a, &b, &c);

		g[a].push_back(make_pair(b,c)); // a -> b with cost c
		g[b].push_back(make_pair(a,c)); // b -> a with cost c (for un-directed graph)
	}

	return 0; 
}

File: Graphs/graph implementation.java

//-------------------------------------------------------------------------------------------
//----------------------------------- unweighted graph --------------------------------------
//-------------------------------------------------------------------------------------------

public class Main {

    static int n,m;
    static List<Integer> [] g;
    
    public static void main(String[] args) throws IOException {
        Scanner in = new Scanner(System.in);
        n = in.nextInt();
        m = in.nextInt();
        
        g = new List[n];
        for(int i=0; i<n; i++)
            g[i] = new ArrayList();
        
        for(int i=0; i<m; i++) {
            int a = in.nextInt();
            int b = in.nextInt();
            
            g[a].add(b); // a -> b
            g[b].add(a); // b -> a  (for un-directed graph)
        }
    } 
}




//-------------------------------------------------------------------------------------------
//------------------------------------- weighted graph --------------------------------------
//-------------------------------------------------------------------------------------------


public class Main {

    static int n,m;
    static List<node> [] g;
    
    public static void main(String[] args) throws IOException {
        Scanner in = new Scanner(System.in);
        n = in.nextInt();
        m = in.nextInt();
        
        g = new List[n];
        for(int i=0; i<n; i++)
            g[i] = new ArrayList();
        
        for(int i=0; i<m; i++) {
            int a = in.nextInt();
            int b = in.nextInt();
            int c = in.nextInt();
            
            g[a].add(new node(b,c)); // a -> b with cost c
            g[b].add(new node(a,c)); // b -> a with cost c (for un directed graph)
        }
    } 
}

class node {
    int u,cost;
    
    public node(int u,int cost) {
        this.u = u;
        this.cost = cost;
    }
}